### PR TITLE
Increase machineSet unit test coverage

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Increase unit test coverage for machineSet controller

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/machineset/controller_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"reflect"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
```
Comment (frobware) [inline pkg/controller/machineset/controller_test.go]: The import ordering looks "wrong" here. Don't we order stdlib at the top?

File: pkg/controller/machineset/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+	for _, tc := range testsCases {
+		got := r.MachineSetToMachines(tc.mapObject)
+		if !reflect.DeepEqual(got, tc.expected) {
+			t.Errorf("Case %s. Got: %v, expected: %v", tc.machine.Name, got, tc.expected)
+		}
+	}
+}
+
+func newFakeClient(initObjects ...runtime.Object) client.Client {
```
Comment (frobware) [inline pkg/controller/machineset/controller_test.go]: Not sure this function buys you much, given it is just a one-line wrapper. And it only appears to be called from one place.

**Comment by davidewatson** [comment]
@frobware is right. This is important. Thanks!

/lgtm

**Comment by enxebre** [comment]
this should be good to go now cc @davidewatson @roberthbailey

**Comment by roberthbailey** [comment]
```
2 files have incorrect boilerplate headers:
I1218 21:29:38.529] pkg/controller/machineset/controller_test.go
I1218 21:29:38.529] pkg/controller/machineset/machine_test.go
```

**Comment by frobware** [comment]
LGTM 

cc @roberthbailey

**Pull request merged**